Switch attribute arguments from field syntax (:) to assignment

monkeydom opened the pull request:
As this seems the agreed upon standard for attribute macros

- colon syntax still works too
- updated all the docs and tests
- added sanity tests for the field syntax
- small typos in docs, made top level docs more consistent

danlehmann reviewed (commented):
This looks great, thanks for doing this. Two things should be added:
- Need some more tests for the old syntax still being valid through tests. Right now, only two tiny tests are duplicated. While we don't need all of them, we should at least have a test each that tests every possible syntax
- Please add a bullet point to CHANGELOG.md

monkeydom commented:
Opted to just keep all the old tests for now and only rename the tests files. Cleaned up all the regular tests to only use the new syntax.
Added a Change log entry and bumped minor version as I consider this a backwards compatible feature.
